tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/ContributionTask.java

```diff
@@ -0,0 +1,500 @@
+package org.graalvm.internal.tck;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.graalvm.internal.tck.model.MetadataIndexEntry;
+import org.graalvm.internal.tck.model.contributing.Question;
+import org.graalvm.internal.tck.utils.ConfigurationStringBuilder;
+import org.graalvm.internal.tck.utils.FilesUtils;
+import org.graalvm.internal.tck.utils.InteractiveTaskUtils;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.FileSystemOperations;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.process.ExecOperations;
+
+import javax.inject.Inject;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.*;
+
+public abstract class ContributionTask extends DefaultTask {
+    private static final String METADATA_INDEX = "metadata/index.json";
+    private static final String BUILD_FILE = "build.gradle";
+    private static final String USER_CODE_FILTER_FILE = "user-code-filter.json";
+    private static final String REQUIRED_DOCKER_IMAGES_FILE = "required-docker-images.txt";
+
+    @Inject
+    protected abstract ExecOperations getExecOperations();
+
+    @Inject
+    protected abstract FileSystemOperations getFileSystemOperations();
+
+    private Path gradlew;
```

Comment:
```suggestion
    private final Path gradlew;
```

Reply:
If I make it final, I can only initialize it immediately in the declaration (but since I am using Gradle's `getProject` to calculate those paths, I would rather call it when the task actually starts `run` function), otherwise I can't modify it in the `initializeWorkingDirectories` function.